sdk/ml/azure-ai-ml/azure/ai/ml/_schema/workspace/networking.py

```diff
@@ -125,9 +182,19 @@ class ManagedNetworkSchema(metaclass=PatchedSchemaMeta):
         casing_transform=camel_to_snake,
         metadata={"description": "isolation mode for the workspace managed network."},
     )
-    outbound_rules = fields.List(NestedField(OutboundRuleSchema, allow_none=False, unknown=EXCLUDE), allow_none=True)
-    network_id = fields.Str(required=False)
-    status = NestedField(ManagedNetworkStatusSchema, dump_only=True)
+    outbound_rules = fields.List(
+        UnionField(
+            [
+                NestedField(FqdnOutboundRuleSchema, allow_none=False, unknown=EXCLUDE),
+                NestedField(ServiceTagOutboundRuleSchema, allow_none=False, unknown=EXCLUDE),
+                NestedField(PrivateEndpointOutboundRuleSchema, allow_none=False, unknown=EXCLUDE),
+            ],
+            allow_none=False,
```

Comment:
The schema will generate Union Fields with the `anyOf` by default. If the user can only provide a YAML matching one of these nested schemas, you can use `oneOf` instead: 
```suggestion
            allow_none=False,
            is_strict=True,
```

Reply:
This also has the added benefit of looking better in intellisense when using VS Code.

Reply:
applied, thanks!